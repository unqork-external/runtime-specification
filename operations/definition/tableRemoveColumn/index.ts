export * from './tableRemoveColumn.model'
export * from './tableRemoveColumn.options'
