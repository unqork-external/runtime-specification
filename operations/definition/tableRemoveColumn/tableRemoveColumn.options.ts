import { Description, Required } from '@tsed/schema'

import { OneOrMany } from '../../../../decorators/schema/oneOrMany.decorator'
import { TrimmedDescription } from '../../../../decorators/schema/trimmedDescription.decorator'

export class TableRemoveColumnOperationOptions {
  @TrimmedDescription(`
    A unique string identifier used to specify the particular table
    or component in the application where the column removal operation is to be executed
  `)
  @Required()
  targetKey: string

  @Description('The column(s) to be removed from the target table ')
  @Required()
  @OneOrMany('string')
  keysToRemove: string | string[]
}
