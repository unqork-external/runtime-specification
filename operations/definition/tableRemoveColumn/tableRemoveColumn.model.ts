import { DiscriminatorValue, Required } from '@tsed/schema'

import { TableRemoveColumnOperationOptions } from './tableRemoveColumn.options'
import { OperationTypes } from '../../enums/operation-types.enum'
import { Operation } from '../../interface/operations.interface'

@DiscriminatorValue(OperationTypes.TABLE_REMOVE_COLUMN)
export class TableRemoveColumnOperation extends Operation<OperationTypes.TABLE_REMOVE_COLUMN> {
  @Required()
  public options: TableRemoveColumnOperationOptions = new TableRemoveColumnOperationOptions()
}
