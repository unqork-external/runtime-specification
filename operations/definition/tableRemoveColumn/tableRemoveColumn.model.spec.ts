import { TableRemoveColumnOperation } from './tableRemoveColumn.model'
import { generateSchemaAndValidate } from '../../../../utilities'

describe('Operation Model: TableRemoveColumn Tests', () => {
  it('TableRemoveColumn Model', () => {
    const { schema } = generateSchemaAndValidate(TableRemoveColumnOperation)

    expect(schema).toMatchSnapshot()
  })
})
