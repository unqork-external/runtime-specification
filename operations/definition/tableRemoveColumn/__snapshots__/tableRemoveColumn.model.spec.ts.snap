// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Operation Model: TableRemoveColumn Tests TableRemoveColumn Model 1`] = `
{
  "definitions": {
    "TableRemoveColumnOperationOptions": {
      "properties": {
        "keysToRemove": {
          "anyOf": [
            {
              "type": "string",
            },
            {
              "items": {
                "type": "string",
              },
              "type": "array",
            },
          ],
          "description": "The column(s) to be removed from the target table ",
        },
        "targetKey": {
          "description": "A unique string identifier used to specify the particular table or component in the application where the column removal operation is to be executed",
          "minLength": 1,
          "type": "string",
        },
      },
      "required": [
        "targetKey",
        "keysToRemove",
      ],
      "type": "object",
    },
  },
  "description": "Removes columns from the Table based on provided column IDs.",
  "properties": {
    "creatorSummary": {
      "description": "A detailed summary of the operation",
      "type": "string",
    },
    "name": {
      "description": "Name of the operation",
      "type": "string",
    },
    "options": {
      "$ref": "#/definitions/TableRemoveColumnOperationOptions",
    },
    "type": {
      "const": "TABLE_REMOVE_COLUMN",
      "examples": [
        "TABLE_REMOVE_COLUMN",
      ],
      "type": "string",
    },
  },
  "required": [
    "options",
  ],
  "type": "object",
}
`;
